### PR TITLE
Feat ammunition save effect

### PR DESCRIPTION
**Pull request by DarKDinDoN**
Took a shot at this.

There is a lot going on in this module. Hope I didn't make a mess.
Made extensive tests, tho !

### CODE REVIEW

**Comment by CarlosFdez** [comment]
Sorry just checked.

Does the active effect work for the last ammo in the pile? I suspect trying to check the item flags at render time will not work if its the last arrow/bullet/whathaveyou.

**Comment by DarKDinDoN** [comment]
@CarlosFdez 
It does. Because using the last ammo with a range weapon doesn't destroy the ammo anyway (even if "destroy on empty" is checked). Didn't modify that part. That's yours :) 

Not sure what would happen if the ammo gets destroy when using a range weapon.
https://github.com/RedReign/FoundryVTT-BetterRolls5e/blob/master/betterrolls5e/scripts/custom-roll.js#L266

To be honest, I don't think the ammo can be destroyed anyway, that would mean a lot of changes for the repeat feature and other stuff.


To summarize. It's working :)

**Comment by CarlosFdez** [comment]
It can be destroyed. Remember to enable auto destroy on the ammo.

**Comment by DarKDinDoN** [comment]
@CarlosFdez not by my tests.
Here is my setup:

image

Ammo doesn't get destroyed after using the last charge with the crossbow.

**Comment by CarlosFdez** [comment]
Just checked again and it seems you're correct about ammo.

Does this also handle active effects on the weapon itself? I'm glancing again and it seems like the item is replaced for some processing, perhaps skipping over any bow/crossbow effects.

If so I'll merge it and do some cleanup.

**Comment by DarKDinDoN** [comment]
@CarlosFdez Hi,
Not sure what you mean by "skipping over any bow/crossbow effects".
But yes, If an effect is declared on the weapon and on the ammo, two effect buttons are displayed in the chat card. And the ammo effect is identified by a special text in the button label "Apply Active Effect [ammo name]"

File: betterrolls5e/scripts/chat-message.js [hunk 1]
```diff
@@ -354,7 +354,10 @@ export class BetterRollsChatCard {
 			}
 
 			const roll = this.roll;
-			const item = await roll.getItem();
+			let item = await roll.getItem();
+			if (data.ammo) {
+				item = item.parent.items.get(item.data.data.consume.target)
```
Comment (CarlosFdez) [inline betterrolls5e/scripts/chat-message.js:359]: What I meant by the comment was that you should probably check that the apply active effects button is for the ammo before doing this replacement. I get the feeling that the button also does AE's for the ammo and never for the weapon if ammo is set.

Unless of course that's what data is.

Comment (CarlosFdez) [reply betterrolls5e/scripts/chat-message.js:359]: That is what it is, got it.

**Review by CarlosFdez — APPROVED** [review]

**Comment by CarlosFdez** [comment]
Alright, looks good to me. Thank you for this!

**Pull request merged**